src/commonAccountFunctions.ts

```diff
@@ -27,27 +27,46 @@ type AddQueryOptionsBase = {
   displayName?: string;
 };
 
-export type AddQueryDataset = AddQueryOptionsBase & { dataset: Dataset; service?: never };
-export type AddQueryService = AddQueryOptionsBase & { service: Service; dataset?: never };
+export type AddQueryDataset = AddQueryOptionsBase & {
+  dataset: Dataset;
+  service?: never;
+  serviceType?: Models.SparqlQueryServiceType;
+};
+export type AddQueryService = AddQueryOptionsBase & { service: Service; dataset?: never; serviceType?: never };
 export async function addQuery<T extends Account>(this: T, name: string, opts: AddQueryDataset): Promise<Query>;
 export async function addQuery<T extends Account>(this: T, name: string, opts: AddQueryService): Promise<Query>;
 export async function addQuery<T extends Account>(this: T, name: string, opts: AddQueryDataset | AddQueryService) {
   const app = (this as User)["_app"];
+  if (!(await app.isCompatible("23.09.0"))) {
+    throw new IncompatibleError(
+      "This function has been updated and is now supported by TriplyDB API version 23.09.0 or greater"
+    );
+  }
   const accountName = (await this.getInfo()).accountName;
   let dataset: string | undefined;
+  let serviceType: Models.SparqlQueryServiceType | undefined;
```

Comment:
@charitha93 Do we need this extra variable? Can't we just use `opts.serviceType` on line 68?